src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -1,15 +1,59 @@
 package TypewiseAlert;
 
+import org.junit.Test;
+
 import static org.junit.Assert.assertTrue;
 
-import org.junit.Test;
+public class TypewiseAlertTest {
+    @Test
+    public void infersBreachTooLow() {
+        assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
+                BreachType.TOO_LOW);
+    }
+
+    @Test
+    public void infersBreachTooHigh() {
+        assertTrue(TypewiseAlert.inferBreach(35, 20, 30) ==
+                BreachType.TOO_HIGH);
+    }
+
+    @Test
+    public void infersBreachNormal() {
+        assertTrue(TypewiseAlert.inferBreach(25, 20, 30) ==
+                BreachType.NORMAL);
+    }
+
+    @Test
+    public void checkNAlert_ToController_Passive_Cooling() {
+        assertTrue(TypewiseAlert.checkAndAlert(AlertTarget.TO_CONTROLLER, CoolingType.PASSIVE_COOLING, 30));
```

Comment:
how do we verify that indeed controller has received the alert? there should be mechanism to check this...